draft-barnes-acme.md

```diff
@@ -444,7 +448,9 @@ The server MUST ignore any values provided in the "key" field in registration bo
 The server creates a registration object with the included contact information.  The "key" element of the registration is set to the public key used to verify the JWS (i.e., the "jwk" element of the JWS header).  The server also provides a random
 recovery token.  The server returns this registration object in a 201 (Created) response, with the registration URI in a Location header field.  The server MUST also indicate its new-authorization URI using the "next" link relation.
 
-If the server wishes to present the client with terms under which the ACME service is to be used, it may indicate the URI where such terms can be accessed in a Link header with link relation "terms-of-service".  As noted above, the client may indicate its
+If the server already has a registration object with the provided account key, then it MUST return a 409 (Conflict) response and provide the URI of that registration in a Location header field.  This allows a client that has an account key but not the corresponding registration URI to recover the registration URI.
```

Comment:
Right. Assuming we decide to solve this (maybe it's fine to issue another one?), instead of looking for duplicate CSRs, perhaps we could look for duplicates on [pubkey, identifiers] created in the last N minutes, and 409 on those.

It's probably worthwhile to change the spec to make cert issuance be async and polling, like solving challenges. I.e. /new-cert/ doesn't return the cert, but returns a reference to a pending cert resource that should be available within ~minutes. Once the cert is ready, that resource would redirect to the final cert URL.